.octopus/deployment_process.ocl

```diff
@@ -0,0 +1,44 @@
+step "run-a-script" {
+    name = "Publish package to Artifactory"
+
+    action "publish-package-to-artifactory" {
+        action_type = "Octopus.Script"
+        properties = {
+            Octopus.Action.Script.ScriptBody = <<-EOT
+                artifactoryUrl="$(get_octopusvariable "ArtifactoryUrl")"
+                packageVersion="$(get_octopusvariable "Octopus.Action.Package[Octopus.Jenkins].PackageVersion")"
+                extractedPath="$(get_octopusvariable "Octopus.Action.Package[Octopus.Jenkins].ExtractedPath")"
+                creds="$(get_octopusvariable "ArtifactoryUsername"):$(get_octopusvariable "ArtifactoryPassword")"
+                publishVersionUrl="$artifactoryUrl/hudson/plugins/octopusdeploy/octopusdeploy/$packageVersion"
+                
+                for filePath in $extractedPath/*
+                do
+                    fileName=$(basename -- $filePath)
+                    publishFileUrl="$publishVersionUrl/$fileName"
+                    echo "Pushing $filePath to $publishFileUrl"
+                    echo '##octopus[stdout-verbose]'
+                    curl -fSs -X PUT \
```

Comment:
Is this just using curl to send to artifactory directly?

Does octopus have a set of artifactory steps? (like the azure devops artifactory extension)
https://www.jfrog.com/confluence/display/JFROG/Artifactory+Azure+DevOps+Extension

Have we considered using the jfrog cli?

This is all big overkill stuff for this PR, but just wondering about the wider world and whether there's value in some tighter artifactory integration

Reply:
> Is this just using curl to send to artifactory directly?

Yes

> Does octopus have a set of artifactory steps?

No

> Have we considered using the jfrog cli?

jfrog cli is not installed by default on any of our workers, so using curl is just easier :)